pkg/daemon/daemon.go

```diff
@@ -236,10 +237,23 @@ func (d *daemon) AddPeriodicUpdate() {
 					glog.Warningf("AddPeriodicUpdate(): failed to release guid \"%s\" from removed pod \"%s\""+
 						" in namespace \"%s\" with error: %v", guidList[index].String(), pod.Name, pod.Namespace, err)
 				}
+
+				removedGuidList = append(removedGuidList, guidList[index])
 			}
 
 			finalPassCounter++
 		}
+
+		if ibCniSpec.PKey != "" && len(removedGuidList) != 0 {
+			// Already check the parse above
+			pKey, _ := utils.ParsePKey(ibCniSpec.PKey)
+			if pkeyErr := d.smClient.RemoveGuidsFromPKey(pKey, removedGuidList); pkeyErr != nil {
+				glog.Errorf("AddPeriodicUpdate(): failed to remove guids of removed pods from pKey %s with subnet manager %s with error: %v",
+					ibCniSpec.PKey, d.smClient.Name(), pkeyErr)
+				continue
```

Comment:
why we need  continue?

Reply:
To prevent removing the network from the map